tests/integration/targets/test_mysql_role/tasks/mysql_role_initial.yml

```diff
@@ -1477,173 +1490,171 @@
     assert:
       that:
-      - result.rowcount.0 == 1
-    when: install_type == 'mariadb'
+        - result.rowcount.0 == 1
+    when: db_engine == 'mariadb'
 
-  - name: Create role with admin again
-    <<: *task_params
+  - name: Create role0 with admin again
     mysql_role:
       <<: *mysql_params
-      name: '{{ role0 }}'
+      name: 'role0'
       state: present
-      admin: '{{ user0 }}@localhost'
+      admin: 'user0@%'
+    register: result
     ignore_errors: yes
 
-  - name: Check with MySQL
+  - name: Assert expected error message in mysql again
     assert:
       that:
-      - result is failed
-      - result.msg is search('option can be used only with MariaDB')
-    when: install_type == 'mysql'
+        - result is failed
+        - result.msg is search('option can be used only with MariaDB')
+    when: db_engine == 'mysql'
 
-  - name: Check with MariaDB
+  - name: Assert create role0 in mariadb is not changed
     assert:
       that:
-      - result is not changed
-    when: install_type == 'mariadb'
+        - result is not changed
+    when: db_engine == 'mariadb'
 
   # Try to grant a role to a user who does not exist
-  - name: Create role with admin again
-    <<: *task_params
+  - name: Create role0 with admin again
     mysql_role:
       <<: *mysql_params
-      name: '{{ role0 }}'
+      name: 'role0'
       state: present
       members:
-      - '{{ nonexistent }}@localhost'
+        - 'nonexistent@%'
+    register: result
     ignore_errors: yes
 
-  - name: Check
+  - name: Assert that create role0 with admin again is failed
     assert:
       that:
-      - result is failed
-      - result.msg is search('does not exist')
+        - result is failed
+        - result.msg is search('does not exist')
 
   always:
-  # Clean up
-  - name: Drop DBs
-    mysql_query:
-      <<: *mysql_params
-      query: 'DROP DATABASE {{ item }}'
-    loop:
-    - '{{ test_db }}'
-    - '{{ test_db1 }}'
-    - '{{ test_db2 }}'
 
-  - name: Drop users
-    <<: *task_params
-    mysql_user:
-      <<: *mysql_params
-      name: '{{ item }}'
-      state: absent
-    loop:
-    - '{{ user0 }}'
-    - '{{ user1 }}'
-    - '{{ user2 }}'
-
-  - name: Drop roles
-    <<: *task_params
-    mysql_role:
-      <<: *mysql_params
-      name: '{{ item }}'
-      state: absent
-    loop:
-    - '{{ role0 }}'
-    - test
-    - '{{ role3 }}'
+    - name: Clean up DBs
+      mysql_query:
+        <<: *mysql_params
+        query: 'DROP DATABASE IF EXISTS {{ item }}'
+      loop:
+        - 'test_db'
+        - 'test_db1'
+        - 'test_db2'
+
+    - name: Clean up users
+      mysql_user:
+        <<: *mysql_params
+        name: '{{ item }}'
+        state: absent
+      loop:
+        - 'user0'
+        - 'user1'
+        - 'user2'
+
+    - name: Clean up roles
+      mysql_role:
+        <<: *mysql_params
+        name: '{{ item }}'
+        state: absent
+      loop:
+        - 'role0'
+        - 'test'
+        - 'role3'
```

Comment:
This is a massive file with a lot of mysql | mariadb tests, we should split it by db_engine. Not part of this PR, though.